app/gameobjects.ts

```diff
@@ -0,0 +1,67 @@
+import * as wire from './wire';
+import RobotProxy from './robotproxy';
+import {GameState} from './gamestate';
+
+export class GameObject {
+    constructor(public location: wire.Point) {}
+    inRange(other: {location: wire.Point}, range: number) {
```

Comment:
Why are we supplying range? wouldn't it be better to have a getDistance() or something?